modules/unsupported/cql-json/src/main/java/org/geotools/filter/text/cqljson/CQLJsonFilterBuilder.java

```diff
@@ -0,0 +1,610 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.filter.text.cqljson;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.geotools.filter.text.cqljson.model.After;
+import org.geotools.filter.text.cqljson.model.Anyinteracts;
+import org.geotools.filter.text.cqljson.model.Before;
+import org.geotools.filter.text.cqljson.model.Begins;
+import org.geotools.filter.text.cqljson.model.Begunby;
+import org.geotools.filter.text.cqljson.model.Between;
+import org.geotools.filter.text.cqljson.model.Contains;
+import org.geotools.filter.text.cqljson.model.Crosses;
+import org.geotools.filter.text.cqljson.model.Disjoint;
+import org.geotools.filter.text.cqljson.model.During;
+import org.geotools.filter.text.cqljson.model.Endedby;
+import org.geotools.filter.text.cqljson.model.Ends;
+import org.geotools.filter.text.cqljson.model.Eq;
+import org.geotools.filter.text.cqljson.model.Equals;
+import org.geotools.filter.text.cqljson.model.FunctionObjectArgument;
+import org.geotools.filter.text.cqljson.model.Gt;
+import org.geotools.filter.text.cqljson.model.Gte;
+import org.geotools.filter.text.cqljson.model.In;
+import org.geotools.filter.text.cqljson.model.Intersects;
+import org.geotools.filter.text.cqljson.model.Like;
+import org.geotools.filter.text.cqljson.model.Lt;
+import org.geotools.filter.text.cqljson.model.Lte;
+import org.geotools.filter.text.cqljson.model.Meets;
+import org.geotools.filter.text.cqljson.model.Metby;
+import org.geotools.filter.text.cqljson.model.Overlappedby;
+import org.geotools.filter.text.cqljson.model.Overlaps;
+import org.geotools.filter.text.cqljson.model.TContains;
+import org.geotools.filter.text.cqljson.model.Tequals;
+import org.geotools.filter.text.cqljson.model.Tintersects;
+import org.geotools.filter.text.cqljson.model.Touches;
+import org.geotools.filter.text.cqljson.model.Toverlaps;
+import org.geotools.filter.text.cqljson.model.Within;
+import org.geotools.filter.text.generated.parsers.ParseException;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+import org.opengis.filter.expression.Literal;
+import org.opengis.filter.expression.PropertyName;
+import org.opengis.geometry.Geometry;
+
+final class CQLJsonFilterBuilder {
```

Comment:
This is a bit odd to have as a builder, are you sure you are not looking for a FilterVisitor? Traverse the data structure and covert ...

Reply:
This is the pattern that the original CQL to filter developers followed.  My understanding is that the FilterVisitor is for going from Filter to something else (like SQL) vs. what I am doing here, which is going from JSON to Filter.